src/libraries/Common/tests/System/Net/Http/Http3LoopbackConnection.cs

```diff
@@ -189,13 +189,36 @@ public override async Task<HttpRequestData> HandleRequestAsync(HttpStatusCode st
 
             await stream.SendResponseAsync(statusCode, headers, content).ConfigureAwait(false);
 
-            // closing the connection here causes bytes written to streams to go missing.
-            // Regardless, we told the client we are closing so it shouldn't matter -- they should not use this connection anymore.
-            //await CloseAsync(H3_NO_ERROR).ConfigureAwait(false);
+            await WaitForClientDisconnectAsync();
 
             return request;
         }
 
+        // Wait for the client to close the connection, e.g. after we send a GOAWAY, or after the HttpClient is disposed.
+        public async Task WaitForClientDisconnectAsync()
+        {
+            while (true)
+            {
+                Http3LoopbackStream stream;
+
+                try
+                {
+                    stream = await AcceptRequestStreamAsync().ConfigureAwait(false);
```

Comment:
Can this end up hanging forever? Should we put a timeout here so we won't have to rely on tests using this to have a timeout

Reply:
Sure, I can add that. Is there a particular timeout we should use here?

Reply:
@wfurt was adding some default timeout constant for Quic tests in #56043
```
public const int PassingTestTimeout = 30_000;
```
Possibly we could add one to HTTP/3 as well and use it here

Reply:
Actually, I'm not going to add this in this PR. We basically don't have any timeout mechanism in Http3LoopbackServer and I don't really want to do a one-off just for this. We should revisit timeouts in the Http3LoopbackServer overall.